Add snap-deal queue (#133)
filecoin-project/lotus-docs: content/en/storage-providers/operate/snap-deals/index.md

> You must have to following set up to follow this guide through:

Reviewer: ```suggestion
You must have the following set up to follow this guide through:
```

    This command does not output anything on success.

> 1. By listing your deals again, you'll see that the FSM has marked the sector as being in a `Available` state:

Reviewer: ```suggestion
1. By listing your deals again, you'll see that the FSM has marked the sector as being in an `Available` state:
```

    ```shell
    lotus-miner sectors list

    > ID  State                 OnChain  Active  Expiration                   Deals  DealWeight  VerifiedPower
    > 0   Proving               YES      YES     1550097 (in 10 weeks 1 day)  CC
    > 1   Proving               YES      YES     1550097 (in 10 weeks 1 day)  1      2KiB        18KiB
    > 2   Available    YES      YES     1549517 (in 10 weeks 1 day)  CC
    ```

    This means that this sector (`2`) is ready to wait for deals!

> While the sector is transitioning through the snap-deals states, this sector is still preserved, and nothing is changing internally. The FSM works to keep the sector safe so that WindowPosts and WinningPosts are totally undisrupted during the process.  

Reviewer: ```suggestion
While the sector is transitioning through the snap-deals states, this sector is still preserved, and nothing is changed internally. The FSM works to keep the sector safe so that WindowPosts and WinningPosts are totally undisrupted during the process.  
```

### Accept the new storage deal

> After the CC sector has been converted to accept the snap-deal, a new storage deal is accepted the storage provider will be sealed into the converted sector. You can track the status by checking the sector logs using the below command.

Reviewer: ```suggestion
After the CC sector has been converted to accept the snap-deal, a new storage deal is accepted by the storage provider and will be sealed into the converted sector. You can track the status by checking the sector logs using the below command.
```

> After the CC sector has been converted to accept the snap-deal, a new storage deal is accepted the storage provider will be sealed into the converted sector. You can track the status by checking the sector logs using the below command.

Reviewer 2: >converted to accept the snap-deal, a new storage deal is accepted the storage provider will be sealed into the converted sector.

Maybe change to:
`converted to accept the snap-deal and a new storage deal is accepted, the storage provider will seal it into the converted sector.`

Author: @rjan90 Can you check this again as I merged the suggestions from @TippyFlitsUK .  This is outdated now.

Reviewer 2: LGTM. Resolving

1. It removes the manual intervention required to convert the "CC" sectors to snap-deal ready sectors and thus, reduces the overhead for the storage provider. 
2. As all the new storage deals will be sealed as part of snap-deal, the sealing time would be considerable less.

This new feature can be enabled in the lotus miner [configuration]({{<relref "../../configure/configuration/#make-new-cc-sector-available-for-snap-deal">}}).

When a deal arrives to the sealing pipeline, lotus-miner will try to match it to any sectors waiting for deal before sealing. If none are found then, lotus-miner will try to check if any `Available` sectors can accept it. It will select a candidate with lowest initial pledge. If none of the sectors are eligible then a new sector can be created based on the value of `MakeNewSectorForDeals` under [configuration]({{<relref "../../configure/configuration/#disabling-new-sector-for-deal">}}). 

{{< alert >}}

> In case creating new sectors for deals is disabled then deal will be left hanging till a sectors is made "Available" for it.

Reviewer: ```suggestion
In the case that creating new sectors for deals is disabled, the deal will be left hanging until a sector is made "Available" for it.
```
